library/extractor/src/main/java/com/google/android/exoplayer2/extractor/ts/TsExtractor.java

```diff
@@ -96,6 +96,9 @@
   public static final int TS_STREAM_TYPE_SPLICE_INFO = 0x86;
   public static final int TS_STREAM_TYPE_DVBSUBS = 0x59;
 
+  //Those are special IDs, which don't have actual TS definitions
```

Comment:
Where does the value come from? Can you reference a spec or something?

Reply:
No, this value is magic value that is beyond max standard value of TS_STREAM_TYPE which is one byte.
It is meant to carry stream type for types not defined in the spec.

Do you have a recommendation to handle that?

Reply:
Sorry to jump in: You can avoid the public magic value by adding a boolean in StreamInfo saying "hasApplicationSignallingDescriptor" which is true if and only if one of the descrptor uses descriptor tag 0x6F. Alternatively, you could include in stream info a list / array of all descriptor tags associated with the program, and then checking for that in the TsPayloadReaderFactory. The boolean is probably just simpler.

Reply:
For what it's worth, Teletext tracks are also detected by a specific descriptor in PMT
(see https://www.etsi.org/deliver/etsi_EN/300400_300499/300472/01.03.01_60/en_300472v010301p.pdf section 4)

So the boolean way doesn't work much. I'm not a fan of including all descriptors tags.
We could have an `enum ExtraType { Nope, Application, Teletext, }` but I'm not much of a fan either.

Now, in Teletext's case, descriptors contain useful information to send to Teletext{Reader,Decoder} (namely the page of the subtitle and the language). At the moment we transmit it by creating a new TelextSubtitleInfo storage-class (which inherits SubtitleInfo, and we changed DvbSubtitleInfo inherits SubtitleInfo as well). And we gave teletext track 0x102 id

I don't have strong opinion on how to handle that, though at the moment I still prefer leaving those "internal" IDs.